### PR TITLE
Store format

### PR DESCRIPTION
**Pull request by alkaitagi**
This document describes the format for the SQLite store by providing event types, their values, and optimization techniques being used.

### CODE REVIEW

File: documents/storage-format.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+### Engagement level change
+
+Interaction states are described in details in [_interaction-states_ document](interaction-states.md). Here we will just assign integer codes to each of them. A record will be made only when the engagement level has changed.
+
+| 0          | 1            | 2      | 3           | 4       |
+| ---------- | ------------ | ------ | ----------- | ------- |
+| engagement | conferencing | idling | distraction | absense |
+
+```SQL
+time: TEXT ("YYYY-MM-DD HH:MM:SS.SSS");
```
Comment (illright) [inline documents/storage-format.md]: In my SQLite store PR I store the time as an integer timestamp. Also maybe it'd be better to include both the format of the SQLite tables and the files?

File: documents/storage-format.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+key_code: INTEGER;
+```
+
+### Cursor position change
+
+```SQL
+time: TEXT ("YYYY-MM-DD HH:MM:SS.SSS");
+cursor_position_x: INTEGER;
```
Comment (illright) [inline documents/storage-format.md]: These names are different from what I have in the SQLite PR

File: documents/storage-format.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+### Button click
+
+| 0          | 1           | 2            |
+| ---------- | ----------- | ------------ |
+| left click | right click | middle click |
+
+```SQL
+time: TEXT ("YYYY-MM-DD HH:MM:SS.SSS");
+key_code: INTEGER;
```
Comment (illright) [inline documents/storage-format.md]: Also, the mouse key codes in my PR are stored as strings

File: documents/storage-format.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+time: TEXT ("YYYY-MM-DD HH:MM:SS.SSS");
+cursor_position_x: INTEGER;
+cursor_position_y: INTEGER;
+```
+
+### Wheel scroll
+
+```SQL
+time: TEXT ("YYYY-MM-DD HH:MM:SS.SSS");
+scroll_delta: REAL;
```
Comment (illright) [inline documents/storage-format.md]: I have this field `integer`, and it seems like the library we're using also outputs integers here. I tried to get a fraction with the touchpad, but it doesn't seem to work with the touchpad. So maybe screw it and just have it integer?

**Review by illright — CHANGES_REQUESTED** [review]

File: documents/storage-format.md (hunk 2 above)
Comment (alkaitagi) [reply documents/storage-format.md]: Renamed to *x* and *y*.

File: documents/storage-format.md (hunk 3 above)
Comment (alkaitagi) [reply documents/storage-format.md]: Changed to *TEXT*.

File: documents/storage-format.md (hunk 4 above)
Comment (alkaitagi) [reply documents/storage-format.md]: Changed to *Integer*.

File: documents/storage-format.md [hunk 5]
```diff
@@ -37,21 +37,21 @@ As of optimization techniques, for continious input channels like _mouse positio
...
+button: TEXT;
 ```
 
 ### Cursor position change
 
 ```SQL
-time: TEXT ("YYYY-MM-DD HH:MM:SS.SSS");
-cursor_position_x: INTEGER;
-cursor_position_y: INTEGER;
+time: INTEGER;
```
Comment (illright) [inline documents/storage-format.md]: This column is called `timestamp`

File: documents/storage-format.md [hunk 6]
```diff
@@ -37,21 +37,21 @@ As of optimization techniques, for continious input channels like _mouse positio
...
+y: INTEGER;
 ```
 
 ### Wheel scroll
 
 ```SQL
-time: TEXT ("YYYY-MM-DD HH:MM:SS.SSS");
-scroll_delta: REAL;
+time: INTEGER;
+delta: INTEGER;
```
Comment (illright) [inline documents/storage-format.md]: This was now renamed to `wheel_delta`, sorry :p

**Review by illright — CHANGES_REQUESTED** [review]

File: documents/storage-format.md (hunk 1 above)
Comment (alkaitagi) [reply documents/storage-format.md]: Included.

File: documents/storage-format.md (hunk 6 above)
Comment (alkaitagi) [reply documents/storage-format.md]: Renamed.

**Review by illright — APPROVED** [review]

**Review by aabounegm — APPROVED** [review]

**Review by kuredoro — APPROVED** [review]

**Review by IamMaxim — APPROVED** [review]

**Pull request merged**